File: OOAD/tic_tac_toe_app/src/com/techlabs/tic/tac/toe/Cell.java
package com.techlabs.tic.tac.toe;

public class Cell {
	private Mark mark;
	
	public Cell() {
		this.mark = Mark.EMPTY;
	}
	
	public Cell(Mark mark) {
		this.mark = mark;
	}
	
	public void setMark(Mark mark) {
		if(this.mark == Mark.EMPTY) {
		this.mark = mark;
	} else {
		throw new RuntimeException("Mark already present");
	}
	}

	public Mark getMark() {
		return mark;
	}

}

File: Go/src/tic_tac_toe_app/components/cell.go
package components

import "fmt"

const (
	NoMark   = "-"
	XMark    = "X"
	OMark    = "O"
	ErrorMsg = "Cell has a mark already.\nTry a different Index."
)

type Cell struct {
	mark string
}

func NewCell() *Cell {
	// &Cell.mark = ""
	return &Cell{
		mark: NoMark,
	}
}

func (cell *Cell) SetMark(mark string) error {
	if cell.mark == NoMark {
		cell.mark = mark
		return nil
	}
	return fmt.Errorf(ErrorMsg)
}

func (cell *Cell) GetMark() string {
	return cell.mark
}
